microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -138,63 +164,208 @@ Your VS Code project should now look somewhat like this:
+      if (fromUnit === 'k') {
+        if (toUnit === 'c') {
+          return value - 273.15;
+        } else if (toUnit === 'f') {
+          return value * 9 / 5 - 459.67;
+        }
+        return value;
+      }
+      throw new Error('Invalid unit');
+    }
+    
+    form.addEventListener('input', () => {
+      const inputTemp = parseFloat(inputField.value);
+      const fromUnit = fromUnitField.value;
+      const toUnit = toUnitField.value;
+    
+      const outputTemp = convertTemp(inputTemp, fromUnit, toUnit);
+      outputField.value = (Math.round(outputTemp * 100) / 100) + ' ' + toUnit.toUpperCase();
+    });
+    ```
+
+1. Open the `index.html` file again and add the following code after the closing `</form>` tag to load the JavaScript file:
+
+    ```html
+    <script src="converter.js"></script>
+    ```
+
+1. Now add some CSS style to the app, to make it more visually interesting. Create a new file called `converter.css` in your project and add the following code to it:
 
-Service workers are defined in a special JavaScript file, described in [Using Service Workers](https://developer.mozilla.org/docs/Web/API/Service_Worker_API/Using_Service_Workers) and [Service Worker API](https://developer.mozilla.org/docs/Web/API/Service_Worker_API).
+    ```css
+    html {
+      background: rgb(243, 243, 243);
+      font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+      font-size: 15pt;
+    }
+    
+    html, body {
+      height: 100%;
+      margin: 0;
+    }
+    
+    body {
+      display: grid;
+      place-items: center;
+    }
+    
+    #converter {
+      width: 15rem;
+      padding: 2rem;
+      border-radius: .5rem;
+      box-shadow: 0 0 2rem 0 #0001;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+    }
+    
+    #converter input, #converter select {
+      font-family: inherit;
+      font-size: inherit;
+      margin-block-end: 1rem;
+      text-align: center;
+      width: 10rem;
+    }
+    
+    #converter #output-temp {
+      font-size: 2rem;
+      font-weight: bold;
+    }
+    ```
 
-To add a service worker in your project:
+1. Open `index.html` again and reference the new CSS file in it by adding the following code inside the `<head>` tag:
+
+    ```html
+    <link rel="stylesheet" href="converter.css">
+    ```
+
+Your VS Code project should now look something like this
+
+![Screenshot of VS Code showing the sample PWA project, with the index.html, converter.js, converter.css, and manifest.json files](../media/visual-studio-project-with-front-end-files.png)
+
+Now browse to `http://localhost:8080` to view your app.
+
+![Running your new PWA, with the frontend code, on localhost.](../media/sample-pwa-app-with-frontend-code.png)
+
+Your app does something useful now but can't be installed yet. You will make your app installable in the next step by creating a service worker.
```

Comment:
```suggestion
Your app does something useful now, but it can't be installed yet. You'll make your app installable in the next step, by creating a service worker.
```